source/device.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <set>
+#include <unordered_map>
+
+namespace sim {
+
+class Link;
+
+enum DeviceType { SWITCH, SENDER, RECEIVER };
+
+class Device {
+public:
+    Device(DeviceType a_type);
+    virtual ~Device();
+    virtual void process();
+
```

Comment:
We discussed methods for updating links
```suggestion
        void add_inlink(Link* link);
        void add_outlink(Link* link, Device* dest);
```

Reply:
Why do we need `Device* dest` in `void add_outlink`? It seems like `link` already contains destination device info

Reply:
You are right, we probably do not need to pass `Device* dest` at the point of creating an out link: the routing table may change, but the link will stay